### PR TITLE
ft: Initial commits to the casbin grpc service

### PR DESCRIPTION
**Pull request by smrpn**
- [x] enforcer and adapter functions implemented. 
- [x] I think the project structure is correct. Will need some advice on that.
- [ ] RBAC skeletons implemented.

### CODE REVIEW

**Comment by hackerchai** [comment]
lgtm

File: src/main.rs [hunk 1]
```diff
@@ -1,3 +1,31 @@
-fn main() {
-    println!("Hello, world!");
+//use casbin_proto::casbin_server::{Casbin, CasbinServer};
+use std::collections::HashMap;
+//use tonic::{transport::Server, Request, Response, Status};
+
+pub mod casbin_proto {
+    tonic::include_proto!("proto");
```
Comment (PsiACE) [inline src/main.rs:6]: Consider outputting the generated rs file  and use it as a module

File: Cargo.toml [hunk 2]
```diff
@@ -7,3 +7,12 @@ edition = "2018"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
+tonic = "0.4.3"
+prost = "0.7.0"
```
Comment (PsiACE) [inline Cargo.toml:11]: Try to bump to tonic 0.5 & prost 0.8

**Comment by hsluoyz** [comment]
@hackerchai @PsiACE plz review

File: src/server/adapter.rs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+pub async fn load_configuration(file: &str) -> Result<Config, std::io::Error> {
+    //Loads a default config from adapter_config in case a custom adapter isn't provided by the client.
+    //DriverName, ConnectionString, and dbSpecified can be configured in the file. Defaults to 'file' mode.
+
+    let config_file = File::open(file).await.expect("file not found");
+    let decoder: Config = serde_json::from_str(file).expect("JSON was not well-formatted");
+    let mut config: Config = Config::default();
+    let re = Regex::new(r"\$\b((\w*))\b");
+
```
Comment (hackerchai) [inline src/server/adapter.rs:64]: Is the regex problem solved

Comment (smrpn) [reply src/server/adapter.rs:64]: nope. I still cannot figure out why it is used here.

**Comment by hsluoyz** [comment]
@hackerchai

**Review by hackerchai — COMMENTED** [review]
lgtm

**Pull request merged**